packages/ai/src/ui/chat-store.ts

```diff
@@ -0,0 +1,518 @@
+import { parsePartialJson } from '../util';
+import type {
+  ReasoningUIPart,
+  TextUIPart,
+  ToolInvocation,
+  ToolInvocationUIPart,
+  UIMessage,
+} from './ui-messages';
+
+interface ChatStoreSubscriber {
+  id: string;
+  onChatChanged: (event: ChatStoreEvent) => void;
+}
+
+export enum ChatStoreEvent {
+  ChatMessagesChanged = 'chat-messages-changed',
+  ChatStatusChanged = 'chat-status-changed',
+  ChatErrorChanged = 'chat-error-changed',
+}
```

Comment:
Avoid enum (string prefered)

Also would it make sense to have a richer event?

```ts
export interface ChatStoreEvent {
  type: 'chat-messages-changed' | 'chat-status-changed' | 'chat-error-changed',
  chatId: number
} 
```

This will also enable us to do discriminated unions later with different information per event type.